platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultMavenArtifactRepository.java

```diff
@@ -165,58 +172,73 @@ public String getDisplayName() {
     }
 
     @Override
-    public URI getUrl() {
+    @ReplacesEagerProperty(replacedAccessors = {
+        @ReplacedAccessor(originalType = URI.class, value = ReplacedAccessor.AccessorType.GETTER, name = "getUrl"),
+    })
+    public Property<URI> getUrl() {
         return urlArtifactRepository.getUrl();
     }
 
-    @Override
-    public void setUrl(URI url) {
-        invalidateDescriptor();
-        urlArtifactRepository.setUrl(url);
+    @Deprecated
+    public void setUrl(String url) {
```

Comment:
❓ The descriptor is no longer invalidated, so it's recreated on read. Unless we have some way of wiring listeners to property setters, I doubt there is a way to fix this.